Marlin/pins.h

```diff
@@ -39,6 +39,8 @@
 
 #if MB(GEN7_CUSTOM)
   #include "pins_GEN7_CUSTOM.h"
+#elif MB(CNControls_12)
```

Comment:
You forgot to define this in the `boards.h` file. And it should be in all-caps by convention. So define it as `BOARD_CNCONTROLS_12`.